### PR TITLE
ci: migrate to github workflow ci

### PR DESCRIPTION
**Pull request by junnplus**
## What type of PR is this? 

- [x] Refactor
- [ ] Feature
- [ ] Bug Fix
- [ ] New Query Runner (Data Source) 
- [ ] New Alert Destination
- [ ] Other

## How is this tested?

- [x] Unit tests (pytest, jest)
- [x] E2E Tests (Cypress)
- [ ] Manually
- [ ] N/A

### CODE REVIEW

**Review by justinclift — APPROVED** [review]

**Comment by justinclift** [comment]
I've just force merged this.  Hopefully that doesn't screw anything up. :grin:

**Comment by justinclift** [comment]
Thanks heaps for being on top of this @junnplus. :smile:

File: client/cypress/cypress.js [hunk 1]
```diff
@@ -63,10 +63,10 @@ function runCypressCI() {
     PERCY_TOKEN_ENCODED,
     CYPRESS_PROJECT_ID_ENCODED,
     CYPRESS_RECORD_KEY_ENCODED,
-    CIRCLE_REPOSITORY_URL,
+    GITHUB_REPOSITORY,
   } = process.env;
 
-  if (CIRCLE_REPOSITORY_URL && CIRCLE_REPOSITORY_URL.includes("getredash/redash")) {
+  if (GITHUB_REPOSITORY === "getredash/redash") {
```
Comment (junnplus) [inline client/cypress/cypress.js:69]: Sorry, it seems that the repository was hard-coded here, causing the e2e test to fail. 😧

Comment (junnplus) [reply client/cypress/cypress.js:69]: https://github.com/RedashCommunity/redash/actions/runs/4932920858/jobs/8816445778

Comment (justinclift) [reply client/cypress/cypress.js:69]: Ahhh.  In that case, do you have time to fix it atm?

I'm currently in the middle of replacing a failed server, so I can't do it just yet. :smile:

Comment (junnplus) [reply client/cypress/cypress.js:69]: It's not difficult to fix it, but I'm not familiar with Cypress. I'm not sure if we need a separate Cypress project.

Comment (junnplus) [reply client/cypress/cypress.js:69]: Cypree's configuration is also hard-coded here:

https://github.com/RedashCommunity/redash/pull/3/files#diff-b803fcb7f17ed9235f1e5cb1fcd2f5d3b2838429d4368ae4c57ce4436577f03fR111-R113

Comment (justinclift) [reply client/cypress/cypress.js:69]: Ahhh ok.  I've used newer versions of Cypress (there were major changes with the v10 release of Cypress, and I'm used to v12).

I can take a look at it tonight, if no-one gets to it before then. :smile:

Comment (junnplus) [reply client/cypress/cypress.js:69]: Perhaps we should move these variables to the secrets configuration in Github.

Comment (justinclift) [reply client/cypress/cypress.js:69]: Yeah, that's a really good idea.  Please do. :smile:

Comment (justinclift) [reply client/cypress/cypress.js:69]: We'll probably need to figure out Percy as well.  I've not touched Percy before.

Comment (justinclift) [reply client/cypress/cypress.js:69]: Looking at this again - a day later - and reading the error output a bit more, it has:

```
You passed the --record flag but this project has not been setup to record.
```

That probably means it's trying to record (video or images) to the "Cypress Cloud".  For our purposes with starting out, we can skip the recording and investigate it later on when needed.

So, it probably means we need to adjust this line:

https://github.com/RedashCommunity/redash/blob/295cc568f5215299410ffa7d4b8382152437e6c6/client/cypress/cypress.js#L82

... to not have `--record` on the end.

I'll try that out now in my personal repo, to see if that gets it working.

If not, then we can more forcefully remove the failing CI bits to "get things working" for now, and add them back later on as we've investigated + fixed them.

Comment (justinclift) [reply client/cypress/cypress.js:69]: Yep, that looks to have been the main problem.  My test of removing the `--record` option in my personal repo worked, so I've just pushed the same thing to our `master` branch.

That should mean our CI tests now work.  In theory. :smile:

Comment (justinclift) [reply client/cypress/cypress.js:69]: It turned out the blank contents for that page were due to either the `Ublock Origin` or `Privacy Badger` plugin.  Disabling both of those temporarily worked.

**Pull request merged**